tests/Unit/TestCreation.hpp

```diff
@@ -63,8 +68,13 @@ T test_creation(const std::string& construction_string) noexcept {
 }
 
 /// \ingroup TestingFrameworkGroup
-/// Construct a factory object from a given string. The string must not contain
-/// the tag name.
+/// Construct a factory object from a given string.
+///
+/// The string must not contain the
+/// name of the option (which is the name of the struct by default if no
+/// `name()` function is present). For example, to create a struct with a
+/// `SizeT` named option the following would be used:
+/// \snippet Test_TestCreation.cpp size_t_argument
```

Comment:
This is factory creation; why not demonstrate the parsing with a derived class? e.g. line 292ish of the test.